diff --git a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT.cpp b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT.cpp
index 1f116f6..eb6e5f9 100644
--- a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT.cpp
+++ b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT.cpp
@@ -16,6 +16,8 @@
  */
 
 #include "UBLOX_AT.h"
+#include "UBLOX_AT_CellularNetwork.h"
+#include "UBLOX_AT_CellularContext.h"
 
 using namespace mbed;
 using namespace events;
@@ -38,6 +40,26 @@ static const intptr_t cellular_properties[AT_CellularBase::PROPERTY_MAX] = {
     0,  // PROPERTY_NON_IP_PDP_TYPE
     1,  // PROPERTY_AT_CGEREP
 };
+//<sophie, porting ublox R410M and N410, add UBLOX_SARA_R410M_02B
+#elif (UBLOX_SARA_R410M_02B == 1) || (UBLOX_SARA_N410 == 1)
+static const intptr_t cellular_properties[AT_CellularBase::PROPERTY_MAX] = {
+    AT_CellularNetwork::RegistrationModeLAC,        // C_EREG
+    AT_CellularNetwork::RegistrationModeDisable,    // C_GREG
+    AT_CellularNetwork::RegistrationModeDisable,    // C_REG
+    0,  // AT_CGSN_WITH_TYPE
+    1,  // AT_CGDATA
+    0,  // AT_CGAUTH
+    1,  // AT_CNMI
+    1,  // AT_CSMP
+    1,  // AT_CMGF
+    1,  // AT_CSDH
+    1,  // PROPERTY_IPV4_STACK
+    0,  // PROPERTY_IPV6_STACK
+    0,  // PROPERTY_IPV4V6_STACK
+    0,  // PROPERTY_NON_IP_PDP_TYPE
+    1,  // PROPERTY_AT_CGEREP
+};
+//>sophie, porting ublox R410M and N410, add UBLOX_SARA_R410M_02B
 #else
 static const intptr_t cellular_properties[AT_CellularBase::PROPERTY_MAX] = {
     AT_CellularNetwork::RegistrationModeDisable,// C_EREG
@@ -80,7 +102,7 @@ CellularDevice *CellularDevice::get_default_instance()
 {
     static UARTSerial serial(MBED_CONF_UBLOX_AT_TX, MBED_CONF_UBLOX_AT_RX, MBED_CONF_UBLOX_AT_BAUDRATE);
 #if defined (MBED_CONF_UBLOX_AT_RTS) && defined(MBED_CONF_UBLOX_AT_CTS)
-    tr_debug("UBLOX_AT flow control: RTS %d CTS %d", MBED_CONF_UBLOX_AT_RTS, MBED_CONF_UBLOX_AT_CTS);
+    printf/*tr_debug*/("UBLOX_AT flow control: RTS %d CTS %d", MBED_CONF_UBLOX_AT_RTS, MBED_CONF_UBLOX_AT_CTS);
     serial.set_flow_control(SerialBase::RTSCTS, MBED_CONF_UBLOX_AT_RTS, MBED_CONF_UBLOX_AT_CTS);
 #endif
     static UBLOX_AT device(&serial);
diff --git a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularContext.cpp b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularContext.cpp
index 04c049e..437eef8 100644
--- a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularContext.cpp
+++ b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularContext.cpp
@@ -51,6 +51,7 @@ void UBLOX_AT_CellularContext::do_connect()
     _cb_data.error = NSAPI_ERROR_NO_CONNECTION;
 
     // Attempt to establish a connection
+/*
 #ifndef TARGET_UBLOX_C030_R41XM
     _cb_data.error = define_context();
 #elif TARGET_UBLOX_C030_R410M
@@ -82,6 +83,13 @@ void UBLOX_AT_CellularContext::do_connect()
         _cb_data.error = NSAPI_ERROR_OK;
     }
 #endif
+*/
+//alvin ublox
+    _is_context_active = true;
+    _is_context_activated = true;
+    _cb_data.error = NSAPI_ERROR_OK;
+//alvin end ublox
+
     if (_cb_data.error != NSAPI_ERROR_OK) {
         // If new PSD context was created and failed to activate, delete it
         if (_new_context_set) {
diff --git a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularNetwork.cpp b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularNetwork.cpp
index 18ddee7..d837c1d 100644
--- a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularNetwork.cpp
+++ b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularNetwork.cpp
@@ -59,7 +59,7 @@ nsapi_error_t UBLOX_AT_CellularNetwork::set_access_technology_impl(RadioAccessTe
         case RAT_HSDPA_HSUPA:
             _at.at_cmd_discard("+URAT", "=", "%d%d", 2, 2);
             break;
-#elif defined(TARGET_UBLOX_C030_R41XM)
+#elif defined(TARGET_UBLOX_C030_R41XM) || (defined(UBLOX_SARA_R410M_02B) && (UBLOX_SARA_R410M_02B == 1))
         case RAT_CATM1:
             _at.at_cmd_discard("+URAT", "=", "%d%d", 7, 8);
             break;
diff --git a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp
index 848fbb5..4507ddf 100644
--- a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp
+++ b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.cpp
@@ -409,6 +409,8 @@ void UBLOX_AT_CellularStack::clear_socket(CellularSocket *socket)
     }
 }
 
+//<sophie, porting ublox R410M and N410, UBLOX_SARA_R410M_02B not support this command
+#if (UBLOX_SARA_R410M_02B == 0) && (UBLOX_SARA_N410 == 0)
 const char *UBLOX_AT_CellularStack::get_ip_address()
 {
     _at.lock();
@@ -471,3 +473,7 @@ nsapi_error_t UBLOX_AT_CellularStack::gethostbyname(const char *host, SocketAddr
 
     return err;
 }
+#endif	// #if (!defined(UBLOX_SARA_R410M_02B) || (UBLOX_SARA_R410M_02B == 0)) && (!defined(UBLOX_SARA_N410) || (UBLOX_SARA_N410 == 0))
+//>sophie, porting ublox R410M and N410, UBLOX_SARA_R410M_02B not support this command
+
+
diff --git a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.h b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.h
index 1614fc7..ef71ac6 100644
--- a/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.h
+++ b/features/cellular/framework/targets/UBLOX/AT/UBLOX_AT_CellularStack.h
@@ -30,10 +30,14 @@ public:
     UBLOX_AT_CellularStack(ATHandler &atHandler, int cid, nsapi_ip_stack_t stack_type);
     virtual ~UBLOX_AT_CellularStack();
 
+	//<sophie, porting ublox R410M and N410, UBLOX_SARA_R410M_02B not support this command
+	#if (UBLOX_SARA_R410M_02B == 0) && (UBLOX_SARA_N410 == 0)
     virtual const char *get_ip_address();
 
     virtual nsapi_error_t gethostbyname(const char *host,
                                         SocketAddress *address, nsapi_version_t version = NSAPI_UNSPEC, const char *interface_name = NULL);
+	#endif
+	//>sophie, porting ublox R410M and N410, UBLOX_SARA_R410M_02B not support this command
 
 protected:
     virtual nsapi_error_t socket_listen(nsapi_socket_t handle, int backlog);
diff --git a/features/cellular/framework/targets/UBLOX/AT/mbed_lib.json b/features/cellular/framework/targets/UBLOX/AT/mbed_lib.json
index 4db0d2f..d1f008a 100644
--- a/features/cellular/framework/targets/UBLOX/AT/mbed_lib.json
+++ b/features/cellular/framework/targets/UBLOX/AT/mbed_lib.json
@@ -3,19 +3,19 @@
     "config": {
         "tx": {
             "help": "TX pin for serial connection. D1 assumed if Arduino Form Factor, needs to be set/overwritten otherwise.",
-            "value": null
+            "value": "PD_5"
         },
         "rx": {
             "help": "RX pin for serial connection. D0 assumed if Arduino Form Factor, needs to be set/overwritten otherwise.",
-            "value": null
+            "value": "PD_6"
         },
         "rts": {
             "help": "RTS pin for serial connection",
-            "value": null
+            "value": "PD_4"
         },
         "cts": {
             "help": "CTS pin for serial connection",
-            "value": null
+            "value": "PD_3"
         },
         "baudrate" : {
             "help": "Serial connection baud rate",
@@ -23,7 +23,7 @@
         },
         "provide-default": {
             "help": "Provide as default CellularDevice [true/false]",
-            "value": false
+            "value": true
         }
     }
 }
diff --git a/targets/targets.json b/targets/targets.json
index 058e9aa..9f99290 100644
--- a/targets/targets.json
+++ b/targets/targets.json
@@ -2877,7 +2877,10 @@
         "device_name": "STM32F429ZI",
         "bootloader_supported": true,
         "overrides": {
-            "network-default-interface-type": "ETHERNET"
+            "network-default-interface-type": "ETHERNET",
+            "tickless-from-us-ticker": false,
+            "deep-sleep-latency": 3,
+            "lpticker_delay_ticks": 4
         }
     },
     "MTB_STM_S2LP": {
